src/main/java/net/snowflake/client/jdbc/SnowflakeResultSetMetaDataV1.java

```diff
@@ -96,7 +97,28 @@ public boolean isAutoIncrement(int column) throws SQLException {
 
   @Override
   public boolean isCaseSensitive(int column) throws SQLException {
-    return false;
+    int colType = getColumnType(column);
+
+    switch (colType) {
+        // Note: SF types ARRAY, OBJECT, BINARY, GEOGRAPHY, GEOMETRY are also represented as
+        // VARCHAR.
+      case Types.VARCHAR:
+      case Types.CHAR:
+      case Types.BINARY:
```

Comment:
Why BINARY is case sensitive? It should be false. Case sensitive means if you use in WHERE clause with data upper, lower or mix case generate different resultset.

Reply:
fixed it, thanks!